src/board_builder/test/accuracy/accuracy_test.py

```diff
@@ -0,0 +1,224 @@
+            aligned_points = [transform_point(point, transformation_matrix) for point in marker.points]
+            for i in range(len(aligned_points)):
+                aligned_points[i] = list(aligned_points[i])
+            aligned_markers.append(
+                Marker(marker_id=marker.marker_id, marker_size=marker.marker_size, points=aligned_points))
+
+        # Return the aligned TargetBoard
+        return TargetBoard(target_id=target_board.target_id, markers=aligned_markers)
+
+    @staticmethod
+    def _calculate_generate_snapshots_inaccuracy(board_definition, snapshots):
+        """
+        Validation method that calculates the difference between the generated board and the original input board
+        """
+        def compare_transforms(T1, T2):
+            return np.linalg.norm(T1 - T2)
+
+        def calculate_relative_transform(marker0_points, marker1_points):
+            def calculate_transform(points):
+                center = np.mean(points, axis=0)
+                v1 = points[1] - points[0]
+                v2 = points[3] - points[0]
+                normal = np.cross(v1, v2)
+                x_axis = v1 / np.linalg.norm(v1)
+                z_axis = normal / np.linalg.norm(normal)
+                y_axis = np.cross(z_axis, x_axis)
+                rotation = np.column_stack((x_axis, y_axis, z_axis))
+                transform = np.eye(4)
+                transform[:3, :3] = rotation
+                transform[:3, 3] = center
+                return transform
+
+            T0 = calculate_transform(marker0_points)
+            T1 = calculate_transform(marker1_points)
+            T0_to_T1 = np.linalg.inv(T0) @ T1
+            return T0_to_T1
+
+        theoretical_transforms = {}
+        for (i, j) in combinations(board_definition.keys(), 2):
+            theoretical_transforms[(i, j)] = calculate_relative_transform(
+                np.array(board_definition[i]), np.array(board_definition[j]))
+
+        # Calculate and compare relative transforms for each snapshot
+        total_difference = 0
+        num_comparisons = 0
+        for snapshot in snapshots:
+            for (i, j) in combinations(snapshot.keys(), 2):
+                snapshot_transform = calculate_relative_transform(
+                    np.array(snapshot[i]), np.array(snapshot[j]))
+                difference = compare_transforms(theoretical_transforms[(i, j)], snapshot_transform)
+                total_difference += difference
+                num_comparisons += 1
+
+        average_difference = total_difference / num_comparisons
+
+        return average_difference
+
+    @staticmethod
+    def _calculate_rms_error_of_two_corner_dataset(simulated_board_arrangement, algorithm_board_arrangement):
+        """
+        Calculates the RMS difference between the output board arrangement with the input board arrangement
+        """
+        def get_corners(target_board):
+            return target_board.get_points()
```

Comment:
The method get_points in TargetBoard looks to me like it is giving me a list of corner points rather than the center point. Please verify. For the second part of the comment, I added some code to sort all the markers before-hand before TargetBoard is even created, so whenever we get a TargetBoard, the order of its markers are already sorted